File: src/main/java/net/slipp/social/security/SlippUser.java
package net.slipp.social.security;

import java.util.Collection;

import net.slipp.domain.ProviderType;

import org.springframework.security.core.GrantedAuthority;
import org.springframework.security.core.userdetails.User;

public class SlippUser extends User {
    private static final long serialVersionUID = -6016106430103939789L;
    
    private ProviderType providerType;

    public SlippUser(String username, String password, ProviderType providerType, Collection<? extends GrantedAuthority> authorities) {
        super(username, password, authorities);
        this.providerType = providerType;
    }
    
    public ProviderType getProviderType() {
        return providerType;
    }
}

File: src/main/scala/net/slipp/social/security/SlippUser.scala
package net.slipp.social.security

import java.util.Collection
import net.slipp.domain.ProviderType
import org.springframework.security.core.GrantedAuthority
import org.springframework.security.core.userdetails.User

class SlippUser(username: String, password: String, providerType: ProviderType, authorities: Collection[_ <: GrantedAuthority])
  extends User(username, password, authorities) {

  def getProviderType: ProviderType = {
    providerType
  }
}
